pool/job.go

```diff
@@ -48,10 +48,10 @@ func NewJob(header string, height uint32) *Job {
 }
 
 // FetchJob fetches the job referenced by the provided id.
-func FetchJob(db *bolt.DB, id string) (*Job, error) {
+func (db *BoltDB) FetchJob(id string) (*Job, error) {
```

Comment:
Perhaps all the `BoltDB` interface implementations need to go into a separate file. Or do you plan on adding the postgres implementations to the struct files as well?

Reply:
That's something we need to discuss. We could take either approach, but tbh I think both are kind of ugly :joy: 

I havent moved the funcs in this PR simply because it would make the changeset massive, and the review painful.

Reply:
Sure thing, let's resolve this in a separate PR.